### PR TITLE
[FlagGems Operator Development Competition] Add median operator

### PR DESCRIPTION
**Pull request by Prakhar54-byte**
- Implement torch.median(input) -> scalar median of all elements
- Implement torch.median(input, dim, keepdim) -> (values, indices)
- Implement median_out and median_dim_values variants
- Uses stable sort for correct tie-breaking behavior
- Supports float16, bfloat16, float32, float64, int32, int64
- 172 tests passing covering 1D-4D tensors, all dtypes, edge cases

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### CODE REVIEW

**Comment by tengqm** [comment]
@Prakhar54-byte Please sign the CLA in order for the team to review your code.

File: src/flag_gems/ops/__init__.py [hunk 1]
```diff
@@ -658,6 +659,10 @@
     "maximum",
     "mean",
     "mean_dim",
+    "median",
+    "median_dim",
+    "median_dim_values",
+    "median_out",
```
Comment (tengqm) [inline src/flag_gems/ops/__init__.py:665]: please
1. register these operators into `conf/operators.yaml`
2. add benchmarks for them under `benchmark/`.

File: src/flag_gems/ops/median.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+"""
+import logging
+import math
+import torch
+
+logger = logging.getLogger(__name__)
+_MedianResult = torch.return_types.median
+
+
+def median(input: torch.Tensor) -> torch.Tensor:
```
Comment (tengqm) [inline src/flag_gems/ops/median.py:88]: We require all operators to be written in Triton.

File: tests/test_median.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    if dtype in INT_DTYPES:
+        return torch.randint(-100, 100, shape, dtype=dtype, device=flag_gems.device)
+    return torch.randn(shape, dtype=dtype, device=flag_gems.device)
+
+
+@pytest.mark.median
+@pytest.mark.parametrize("n", SHAPES_1D)
+@pytest.mark.parametrize("dtype", ALL_DTYPES)
+def test_median_scalar(dtype, n):
+    from flag_gems.ops.median import median
```
Comment (tengqm) [inline tests/test_median.py]: please avoid imports in the middle of the program, use

```
from flag_gems.ops import median
```

or other ways to enable flaggems.

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
